src/en/troubleshooting-upgrade.md

```diff
@@ -94,3 +94,21 @@ juju upgrade-juju --ignore-agent-versions
 !!! Note:
     The flag `--ignore-agent-versions` is only available starting with Juju
     2.2.6.
+
+
+## Case #4 - Recovering from an upgrade failure
+
+If an attempted upgrade results in failure it may prove difficult to return to
+a working setup. The first remediation plan is to remove the controller with
```

Comment:
I think this probably warrants some further detail - i.e. this might get Juju working again, but will lose anything in that cloud for sure. Also 'remediation plan'  doesn't sound very 'plain English'

Reply:
Right, I can underline that the original controller is not actually being recovered.

OK, for 'remediation plan'.